diff --git a/CMake/FollyCompilerMSVC.cmake b/CMake/FollyCompilerMSVC.cmake
index ad03039aa..1c76e0f00 100644
--- a/CMake/FollyCompilerMSVC.cmake
+++ b/CMake/FollyCompilerMSVC.cmake
@@ -289,6 +289,7 @@ function(apply_folly_compile_options_to_target THETARGET)
   # And the extra defines:
   target_compile_definitions(${THETARGET}
     PUBLIC
+      NOMINMAX
       _CRT_NONSTDC_NO_WARNINGS # Don't deprecate posix names of functions.
       _CRT_SECURE_NO_WARNINGS # Don't deprecate the non _s versions of various standard library functions, because safety is for chumps.
       _SCL_SECURE_NO_WARNINGS # Don't deprecate the non _s versions of various standard library functions, because safety is for chumps.
diff --git a/CMake/FollyCompilerUnix.cmake b/CMake/FollyCompilerUnix.cmake
index f10e1c320..501f55fbd 100644
--- a/CMake/FollyCompilerUnix.cmake
+++ b/CMake/FollyCompilerUnix.cmake
@@ -28,9 +28,9 @@ set(
 )
 mark_as_advanced(CXX_STD)
 
-set(CMAKE_CXX_FLAGS_COMMON "-g -Wall -Wextra")
+set(CMAKE_CXX_FLAGS_COMMON "-Wall -Wextra")
 set(CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} ${CMAKE_CXX_FLAGS_COMMON}")
-set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} ${CMAKE_CXX_FLAGS_COMMON} -O3")
+set(CMAKE_CXX_FLAGS_RELEASE "${CMAKE_CXX_FLAGS_RELEASE} ${CMAKE_CXX_FLAGS_COMMON}")
 
 # Note that CMAKE_REQUIRED_FLAGS must be a string, not a list
 set(CMAKE_REQUIRED_FLAGS "${CMAKE_REQUIRED_FLAGS} -std=${CXX_STD}")
@@ -43,7 +43,6 @@ function(apply_folly_compile_options_to_target THETARGET)
   )
   target_compile_options(${THETARGET}
     PRIVATE
-      -g
       -std=${CXX_STD}
       -finput-charset=UTF-8
       -fsigned-char
diff --git a/CMake/folly-deps.cmake b/CMake/folly-deps.cmake
index 989259a87..c00715a49 100644
--- a/CMake/folly-deps.cmake
+++ b/CMake/folly-deps.cmake
@@ -35,7 +35,7 @@ else()
 endif()
 set(Boost_USE_STATIC_LIBS "${FOLLY_BOOST_LINK_STATIC}")
 
-find_package(Boost 1.51.0 MODULE
+find_package(Boost  
   COMPONENTS
     context
     filesystem
@@ -48,27 +48,28 @@ find_package(Boost 1.51.0 MODULE
 list(APPEND FOLLY_LINK_LIBRARIES ${Boost_LIBRARIES})
 list(APPEND FOLLY_INCLUDE_DIRECTORIES ${Boost_INCLUDE_DIRS})
 
-find_package(DoubleConversion MODULE REQUIRED)
-list(APPEND FOLLY_LINK_LIBRARIES ${DOUBLE_CONVERSION_LIBRARY})
-list(APPEND FOLLY_INCLUDE_DIRECTORIES ${DOUBLE_CONVERSION_INCLUDE_DIR})
-
-find_package(Gflags MODULE)
-set(FOLLY_HAVE_LIBGFLAGS ${LIBGFLAGS_FOUND})
-if(LIBGFLAGS_FOUND)
-  list(APPEND FOLLY_LINK_LIBRARIES ${LIBGFLAGS_LIBRARY})
-  list(APPEND FOLLY_INCLUDE_DIRECTORIES ${LIBGFLAGS_INCLUDE_DIR})
-  set(FOLLY_LIBGFLAGS_LIBRARY ${LIBGFLAGS_LIBRARY})
-  set(FOLLY_LIBGFLAGS_INCLUDE ${LIBGFLAGS_INCLUDE_DIR})
+find_package(double-conversion  REQUIRED)
+list(APPEND FOLLY_LINK_LIBRARIES ${double-conversion_LIBRARY})
+list(APPEND FOLLY_INCLUDE_DIRECTORIES ${double-conversion_INCLUDE_DIR})
+
+find_package(gflags  REQUIRED)
+set(FOLLY_HAVE_LIBGFLAGS ${gflags_FOUND})
+if(gflags_FOUND)
+  list(APPEND FOLLY_LINK_LIBRARIES ${gflags_LIBRARIES})
+  list(APPEND FOLLY_INCLUDE_DIRECTORIES ${gflags_INCLUDE_DIR})
+  set(FOLLY_LIBGFLAGS_LIBRARY ${gflags_LIBRARIES})
+  set(FOLLY_LIBGFLAGS_INCLUDE ${gflags_INCLUDE_DIR})
 endif()
 
-find_package(Glog MODULE)
-set(FOLLY_HAVE_LIBGLOG ${GLOG_FOUND})
-list(APPEND FOLLY_LINK_LIBRARIES ${GLOG_LIBRARY})
-list(APPEND FOLLY_INCLUDE_DIRECTORIES ${GLOG_INCLUDE_DIR})
+find_package(glog  REQUIRED)
+set(FOLLY_HAVE_LIBGLOG ${glog_FOUND})
+list(APPEND FOLLY_LINK_LIBRARIES ${glog_LIBRARY})
+list(APPEND FOLLY_INCLUDE_DIRECTORIES ${glog_INCLUDE_DIR})
 
-find_package(LibEvent MODULE REQUIRED)
-list(APPEND FOLLY_LINK_LIBRARIES ${LIBEVENT_LIB})
-list(APPEND FOLLY_INCLUDE_DIRECTORIES ${LIBEVENT_INCLUDE_DIR})
+
+find_package(Libevent  REQUIRED)
+list(APPEND FOLLY_LINK_LIBRARIES ${libevent_LIBS})
+list(APPEND FOLLY_INCLUDE_DIRECTORIES ${libevent_INCLUDE_DIR})
 
 find_package(ZLIB MODULE)
 set(FOLLY_HAVE_LIBZ ${ZLIB_FOUND})
@@ -79,22 +80,22 @@ if (ZLIB_FOUND)
 endif()
 
 find_package(OpenSSL 1.1.1 MODULE REQUIRED)
-list(APPEND FOLLY_LINK_LIBRARIES ${OPENSSL_LIBRARIES})
-list(APPEND FOLLY_INCLUDE_DIRECTORIES ${OPENSSL_INCLUDE_DIR})
-list(APPEND CMAKE_REQUIRED_LIBRARIES ${OPENSSL_LIBRARIES})
-list(APPEND CMAKE_REQUIRED_INCLUDES ${OPENSSL_INCLUDE_DIR})
+list(APPEND FOLLY_LINK_LIBRARIES ${OpenSSL_LIBRARIES})
+list(APPEND FOLLY_INCLUDE_DIRECTORIES ${OpenSSL_INCLUDE_DIR})
+list(APPEND CMAKE_REQUIRED_LIBRARIES ${OpenSSL_LIBRARIES})
+list(APPEND CMAKE_REQUIRED_INCLUDES ${OpenSSL_INCLUDE_DIR})
 check_function_exists(ASN1_TIME_diff FOLLY_HAVE_OPENSSL_ASN1_TIME_DIFF)
-list(REMOVE_ITEM CMAKE_REQUIRED_LIBRARIES ${OPENSSL_LIBRARIES})
-list(REMOVE_ITEM CMAKE_REQUIRED_INCLUDES ${OPENSSL_INCLUDE_DIR})
+list(REMOVE_ITEM CMAKE_REQUIRED_LIBRARIES ${OpenSSL_LIBRARIES})
+list(REMOVE_ITEM CMAKE_REQUIRED_INCLUDES ${OpenSSL_INCLUDE_DIR})
 if (ZLIB_FOUND)
     list(REMOVE_ITEM CMAKE_REQUIRED_LIBRARIES ${ZLIB_LIBRARIES})
 endif()
 
 find_package(BZip2 MODULE)
-set(FOLLY_HAVE_LIBBZ2 ${BZIP2_FOUND})
-if (BZIP2_FOUND)
-  list(APPEND FOLLY_INCLUDE_DIRECTORIES ${BZIP2_INCLUDE_DIRS})
-  list(APPEND FOLLY_LINK_LIBRARIES ${BZIP2_LIBRARIES})
+set(FOLLY_HAVE_LIBBZ2 ${BZip2_FOUND})
+if (BZip2_FOUND)
+  list(APPEND FOLLY_INCLUDE_DIRECTORIES ${BZip2_INCLUDE_DIRS})
+  list(APPEND FOLLY_LINK_LIBRARIES ${BZip2_LIBRARIES})
 endif()
 
 find_package(LibLZMA MODULE)
@@ -104,32 +105,32 @@ if (LIBLZMA_FOUND)
   list(APPEND FOLLY_LINK_LIBRARIES ${LIBLZMA_LIBRARIES})
 endif()
 
-find_package(LZ4 MODULE)
-set(FOLLY_HAVE_LIBLZ4 ${LZ4_FOUND})
-if (LZ4_FOUND)
-  list(APPEND FOLLY_INCLUDE_DIRECTORIES ${LZ4_INCLUDE_DIR})
-  list(APPEND FOLLY_LINK_LIBRARIES ${LZ4_LIBRARY})
+find_package(lz4 MODULE)
+set(FOLLY_HAVE_LIBLZ4 ${lz4_FOUND})
+if (lz4_FOUND)
+  list(APPEND FOLLY_INCLUDE_DIRECTORIES ${lz4_INCLUDE_DIR})
+  list(APPEND FOLLY_LINK_LIBRARIES ${lz4_LIBRARY})
 endif()
 
-find_package(Zstd MODULE)
-set(FOLLY_HAVE_LIBZSTD ${ZSTD_FOUND})
-if(ZSTD_FOUND)
-  list(APPEND FOLLY_INCLUDE_DIRECTORIES ${ZSTD_INCLUDE_DIR})
-  list(APPEND FOLLY_LINK_LIBRARIES ${ZSTD_LIBRARY})
+find_package(zstd MODULE)
+set(FOLLY_HAVE_LIBZSTD ${zstd_FOUND})
+if(zstd_FOUND)
+  list(APPEND FOLLY_INCLUDE_DIRECTORIES ${zstd_INCLUDE_DIR})
+  list(APPEND FOLLY_LINK_LIBRARIES ${zstd_LIBRARY})
 endif()
 
 find_package(Snappy MODULE)
 set(FOLLY_HAVE_LIBSNAPPY ${SNAPPY_FOUND})
 if (SNAPPY_FOUND)
-  list(APPEND FOLLY_INCLUDE_DIRECTORIES ${SNAPPY_INCLUDE_DIR})
-  list(APPEND FOLLY_LINK_LIBRARIES ${SNAPPY_LIBRARY})
+  list(APPEND FOLLY_INCLUDE_DIRECTORIES ${Snappy_INCLUDE_DIR})
+  list(APPEND FOLLY_LINK_LIBRARIES ${Snappy_LIBRARY})
 endif()
 
-find_package(LibDwarf)
-list(APPEND FOLLY_LINK_LIBRARIES ${LIBDWARF_LIBRARIES})
-list(APPEND FOLLY_INCLUDE_DIRECTORIES ${LIBDWARF_INCLUDE_DIRS})
+find_package(libdwarf)
+list(APPEND FOLLY_LINK_LIBRARIES ${libdwarf_LIBRARIES})
+list(APPEND FOLLY_INCLUDE_DIRECTORIES ${libdwarf_INCLUDE_DIRS})
 
-find_package(Libiberty)
+find_package(libiberty)
 list(APPEND FOLLY_LINK_LIBRARIES ${LIBIBERTY_LIBRARIES})
 list(APPEND FOLLY_INCLUDE_DIRECTORIES ${LIBIBERTY_INCLUDE_DIRS})
 
@@ -141,9 +142,9 @@ find_package(LibUring)
 list(APPEND FOLLY_LINK_LIBRARIES ${LIBURING_LIBRARIES})
 list(APPEND FOLLY_INCLUDE_DIRECTORIES ${LIBURING_INCLUDE_DIRS})
 
-find_package(Libsodium)
-list(APPEND FOLLY_LINK_LIBRARIES ${LIBSODIUM_LIBRARIES})
-list(APPEND FOLLY_INCLUDE_DIRECTORIES ${LIBSODIUM_INCLUDE_DIRS})
+find_package(libsodium)
+list(APPEND FOLLY_LINK_LIBRARIES ${libsodium_LIBRARIES})
+list(APPEND FOLLY_INCLUDE_DIRECTORIES ${libsodium_INCLUDE_DIRS})
 
 list(APPEND FOLLY_LINK_LIBRARIES ${CMAKE_DL_LIBS})
 list(APPEND CMAKE_REQUIRED_LIBRARIES ${CMAKE_DL_LIBS})
@@ -153,10 +154,10 @@ if (PYTHON_EXTENSIONS)
   find_package(Cython 0.26 REQUIRED)
 endif ()
 
-find_package(LibUnwind)
-list(APPEND FOLLY_LINK_LIBRARIES ${LIBUNWIND_LIBRARIES})
-list(APPEND FOLLY_INCLUDE_DIRECTORIES ${LIBUNWIND_INCLUDE_DIRS})
-if (LIBUNWIND_FOUND)
+find_package(libunwind)
+list(APPEND FOLLY_LINK_LIBRARIES ${libunwind_LIBRARIES})
+list(APPEND FOLLY_INCLUDE_DIRECTORIES ${libunwind_INCLUDE_DIRS})
+if (libunwind_FOUND)
   set(FOLLY_HAVE_LIBUNWIND ON)
 endif()
 if (CMAKE_SYSTEM_NAME MATCHES "FreeBSD")
@@ -173,7 +174,7 @@ CHECK_INCLUDE_FILE_CXX(elf.h FOLLY_HAVE_ELF)
 find_package(Backtrace)
 
 set(FOLLY_HAVE_BACKTRACE ${Backtrace_FOUND})
-set(FOLLY_HAVE_DWARF ${LIBDWARF_FOUND})
+set(FOLLY_HAVE_DWARF ${libdwarf_FOUND})
 if (NOT WIN32 AND NOT APPLE)
   set(FOLLY_USE_SYMBOLIZER ON)
 endif()
diff --git a/CMakeLists.txt b/CMakeLists.txt
index ddfdc766e..483cf94bc 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -29,12 +29,13 @@ endif()
 
 # includes
 set(CMAKE_MODULE_PATH
+  ${CMAKE_MODULE_PATH}
   "${CMAKE_CURRENT_SOURCE_DIR}/CMake"
   # for in-fbsource builds
   "${CMAKE_CURRENT_SOURCE_DIR}/../opensource/fbcode_builder/CMake"
   # For shipit-transformed builds
   "${CMAKE_CURRENT_SOURCE_DIR}/build/fbcode_builder/CMake"
-  ${CMAKE_MODULE_PATH})
+  )
 
 # package information
 set(PACKAGE_NAME      "folly")
diff --git a/folly/CMakeLists.txt b/folly/CMakeLists.txt
index e0e16df12..27d4e52c5 100644
--- a/folly/CMakeLists.txt
+++ b/folly/CMakeLists.txt
@@ -28,7 +28,6 @@ install(
 )
 
 add_subdirectory(experimental/exception_tracer)
-add_subdirectory(logging/example)
 
 if (PYTHON_EXTENSIONS)
   # Create tree of symbolic links in structure required for successful
